### PR TITLE
fixed statefulset pvc mount

### PR DESCRIPTION
**Pull request by AndrewChubatiuk**
Fixes https://github.com/VictoriaMetrics/helm-charts/issues/189

### CODE REVIEW

**Review by zekker6 — APPROVED** [review]

File: charts/victoria-metrics-single/templates/server-statefulset.yaml [hunk 1]
```diff
@@ -257,16 +257,13 @@ spec:
           configMap:
             name: {{ include "victoria-metrics.server.relabel.configname" . }}
         {{- end }}
-        {{- if not $app.persistentVolume.enabled }}
-        - name: server-volume
-          emptyDir: {{ toYaml $app.emptyDir | nindent 12 }}
-        {{- else }}
-        {{- if $app.persistentVolume.existingClaim }}
         - name: server-volume
+          {{- if $app.persistentVolume.enabled }}
```
Comment (mkilchhofer) [inline charts/victoria-metrics-single/templates/server-statefulset.yaml:261]: breaking change

File: charts/victoria-metrics-single/templates/server-statefulset.yaml [hunk 2]
```diff
@@ -287,7 +284,7 @@ spec:
         resources:
           requests:
             storage: {{ $app.persistentVolume.size }}
-        {{- with $app.persistentVolume.storageClass }}
+        {{- with $app.persistentVolume.storageClassName }}
```
Comment (mkilchhofer) [inline charts/victoria-metrics-single/templates/server-statefulset.yaml:287]: breaking change

**Review by mkilchhofer — COMMENTED** [review]
Hi @AndrewChubatiuk 

You introduced a breaking change not honoring the SemVer specification.

grafik

Renovate updated my helm chart version from 0.10.1 -> 0.11.0:
-https://github.com/mkilchhofer/k8s-gitops-services/pull/315

In my opinion the chart is now broken:
- If you use the volumeClaimTemplate within a StatefulSet, you should not add the same volume in the volume section again.

**Comment by AndrewChubatiuk** [comment]
het @mkilchhofer 
understand your concern, but chart major version is `0`
also thanks for finding, addressed PVC issue in #1402

**Comment by AndrewChubatiuk** [comment]
fix should be available in [0.11.1](https://github.com/VictoriaMetrics/helm-charts/releases/tag/victoria-metrics-single-0.11.1)

**Comment by mkilchhofer** [comment]
Thanks a lot @AndrewChubatiuk I will test it asap :)

**Comment by mkilchhofer** [comment]
> Thanks a lot @AndrewChubatiuk I will test it asap :)

@AndrewChubatiuk I can confirm that chart version 0.11.1 works (when also applying the breaking change `storageClass` -> `storageClassName`).

**Pull request merged**